realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -0,0 +1,225 @@
+                        realm.removeChangeListener(listener);
+                    }
+                }));
+                subscriber.onNext(realm);
+            }
+        });
+    }
+
+    @Override
+    public Observable<DynamicRealm> from(final DynamicRealm realm) {
+        checkRxJavaAvailable();
+        return Observable.create(new Observable.OnSubscribe<DynamicRealm>() {
+            @Override
+            public void call(final Subscriber<? super DynamicRealm> subscriber) {
+                final RealmChangeListener listener = new RealmChangeListener() {
+                    @Override
+                    public void onChange() {
+                        if (!subscriber.isUnsubscribed()) {
+                            subscriber.onNext(realm);
+                        }
+                    }
+                };
+                realm.addChangeListener(listener);
+                subscriber.add(Subscriptions.create(new Action0() {
+                    @Override
+                    public void call() {
+                        realm.removeChangeListener(listener);
+                    }
+                }));
+
+                // Immediately call onNext with the current value, as due to Realms auto-update, it will be the latest
+                // value.
+                subscriber.onNext(realm);
+            }
+        });
+    }
+
+    @Override
+    public <E extends RealmObject> Observable<RealmResults<E>> from(Realm realm, RealmResults<E> results) {
+        checkRxJavaAvailable();
+        return getRealmResultsObservable(results);
+    }
+
+    @Override
+    public Observable<RealmResults<DynamicRealmObject>> from(DynamicRealm realm, RealmResults<DynamicRealmObject> results) {
+        checkRxJavaAvailable();
+        return getRealmResultsObservable(results);
+    }
+
+    private <E extends RealmObject> Observable<RealmResults<E>> getRealmResultsObservable(final RealmResults<E> results) {
+        return Observable.create(new Observable.OnSubscribe<RealmResults<E>>() {
+            @Override
+            public void call(final Subscriber<? super RealmResults<E>> subscriber) {
+                final RealmChangeListener listener = new RealmChangeListener() {
+                    @Override
+                    public void onChange() {
+                        if (!subscriber.isUnsubscribed()) {
+                            subscriber.onNext(results);
+                        }
+                    }
+                };
+                results.addChangeListener(listener);
+                subscriber.add(Subscriptions.create(new Action0() {
+                    @Override
+                    public void call() {
+                        results.removeChangeListener(listener);
+                    }
+                }));
+
+                // Immediately call onNext with the current value, as due to Realms auto-update, it will be the latest
+                // value.
+                subscriber.onNext(results);
```

Comment:
You can also check `subscriber.isUnsubscribed()` here as well or drop the `if` on line 128 depending on how eagerly you want to handle the small removeChangeListener/unsubscription window.